### PR TITLE
[docs] add page on iOS 16 developer mode

### PR DESCRIPTION
**Pull request by esamelson**
# Why

part of ENG-5786 -- adds a docs page on this flow in addition to the changes that will be made to the build page on the website

# How

Went through the flow of enabling Developer Mode on iOS 16, took screenshots and wrote up instructions.

Especially interested in feedback on:
- whether screenshots (and the page as a whole?) are necessary
- where this should live (I put it in Guides -- it applies to EAS Build but also development builds and anytime you run expo run:ios)

# Test Plan

Ensured the page renders locally as expected:

Screen Shot 2022-09-08 at 5 23 28 PM

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/guides/ios-developer-mode.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import ImageSpotlight from '~/components/plugins/ImageSpotlight'
+
+On devices running iOS 16 and above, you will need to enable a special OS-level "Developer Mode" before you can run [internal distribution](../build/internal-distribution) builds (including those built with EAS) or local development builds installed on the device.
+
+This does not apply to builds signed using enterprise provisioning, nor to any builds installed on an iOS simulator.
+
+To enable Developer Mode on your iOS 16+ device, follow these instructions (once per device):
+
+- In the **Settings** app, navigate to **Privacy & Security > Developer Mode**.
```
Comment (jonsamp) [inline docs/pages/guides/ios-developer-mode.md:13]:
```suggestion
- In the Settings app, navigate to "Privacy & Security" > "Developer Mode".
```

File: docs/pages/guides/ios-developer-mode.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+<ImageSpotlight alt="Navigating to Developer Mode setting" src="/static/images/ios-16-developer-mode-1.png" style={{maxWidth: 480}} />
+
+- Enable the toggle if it isn't already on. You will receive a prompt from iOS to restart your device. Press **Restart** to do so.
```
Comment (jonsamp) [inline docs/pages/guides/ios-developer-mode.md:17]:
```suggestion
- Enable the toggle if it isn't already on. You will receive a prompt from iOS to restart your device. Press "Restart" to do so.
```

File: docs/pages/guides/ios-developer-mode.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+<ImageSpotlight alt="Developer Mode restart prompt" src="/static/images/ios-16-developer-mode-2.png" style={{maxWidth: 480}} />
+
+- After the device restarts, unlock your device; a system alert should appear. Press **Turn On** and then, when prompted, enter your passcode.
```
Comment (jonsamp) [inline docs/pages/guides/ios-developer-mode.md:21]:
```suggestion
- After the device restarts, unlock your device; a system alert should appear. Press "Turn On" and then, when prompted, enter your passcode.
```

**Review by jonsamp — APPROVED** [review]
Thanks for adding this. For now, Guides is a good spot to put this. Also, the screenshots are great and make this a lot more straightforward. Appreciate you thinking about user-flows and helping people see what they'll need to do. When I ran into this myself, I wish a doc like this would have existed.

**Review by brentvatne — APPROVED** [review]

**Comment by brentvatne** [comment]
thanks for this @esamelson! i went ahead and applied @jonsamp's suggestions, rebased, and landed this since ios 16 is out now and this is valuable to have out there

**Pull request merged**